server/src/test/java/io/crate/integrationtests/UserDefinedFunctionsIntegrationTest.java

```diff
@@ -344,4 +349,20 @@ public void test_pg_function_is_visible_when_oid_is_retrieved_from_column() thro
         assertThat(response.rows()[0][0], is(true));
         execute("drop table oid_test");
     }
+
+    @Test
+    public void test_udf_used_inside_generated_column_definition_cannot_be_dropped() {
+        execute("create function doc.foo(long) returns string language dummy_lang as" +
+            " 'function foo(a) { return a; }'");
+        execute("create table doc.t1 (id long, l as doc.foo(id))");
+
+        assertThrows(
+            () -> execute("drop function doc.foo(long)"),
+            isSQLError(containsString(
+                    "Function 'doc.foo(bigint)' cannot be dropped because it is still in use as a generated column expression at: doc.t1.l AS doc.foo(id)"),
+                INTERNAL_ERROR,
+                BAD_REQUEST,
+                4003)
```

Comment:
We must pass in the schemas here instead of getting it as a guice singleton on the ctor to avoid circular dependencies.